api/types/src/transaction.rs

```diff
@@ -65,6 +66,33 @@ impl From<(TransactionWithProof, aptos_crypto::HashValue)> for TransactionOnChai
             info: txn.proof.transaction_info,
             events: txn.events.unwrap_or_default(),
             accumulator_root_hash,
+            changes: Default::default(),
+        }
+    }
+}
+
+impl
+    From<(
+        TransactionWithProof,
+        aptos_crypto::HashValue,
+        TransactionOutputListWithProof,
+    )> for TransactionOnChainData
+{
+    fn from(
+        (txn, accumulator_root_hash, txn_output_list): (
+            TransactionWithProof,
+            aptos_crypto::HashValue,
+            TransactionOutputListWithProof,
+        ),
+    ) -> Self {
+        let (_, txn_output) = &txn_output_list.transactions_and_outputs[0];
+        Self {
```

Comment:
this feels hacky, since you could theoretically have 0, 1, or many outputs.

Reply:
the type is `Vec<(Transaction, TransactionOutput)>` - given we have one transaction here, there should only ever be one value in this array

Reply:
Considering one can do whatever they want and call this function.... stupidity reigns supreme!!! why not just get the txn_output and call this?